azext_iot/tests/digitaltwins/test_dt_resource_unit.py

```diff
@@ -0,0 +1,166 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import re
+import pytest
+import responses
+import json
+import azext_iot.digitaltwins.providers.resource
+from time import sleep
+from azext_iot.digitaltwins import commands_resource as subject
+from azext_iot.tests.digitaltwins.dt_helpers import generate_generic_id
+from msrestazure.azure_exceptions import CloudError
+from azext_iot.digitaltwins.common import ADTPublicNetworkAccessType
+
+# Default values
+name = generate_generic_id()
+resource_group_name = generate_generic_id()
+location = 'westus'
+role_type = "Contributor"
+public_network_access = ADTPublicNetworkAccessType.enabled.value
+provisioning = json.dumps({"provisioningState": "provisioning"})
+finished = json.dumps({"provisioningState": "succeeded"})
+failed = json.dumps({"provisioningState": "failed"})
+
+
+@pytest.fixture
+def start_twin_response(mocked_response, fixture_dt_client):
+    mocked_response.assert_all_requests_are_fired = False
+
+    mocked_response.add(
+        method=responses.GET,
+        content_type="application/json",
+        url=re.compile(
+            "https://management.azure.com/subscriptions/(.*)/resourcegroups/(.*)?"
+        ),
+        status=200,
+        match_querystring=False,
+        body=json.dumps({"name": name}),
+    )
+
+    mocked_response.add(
+        method=responses.GET,
+        content_type="application/json",
+        url=re.compile(
+            "https://management.azure.com/subscriptions/(.*)/"
+            "providers/Microsoft.DigitalTwins/digitalTwinsInstances"
+        ),
+        status=200,
+        match_querystring=False,
+        body=json.dumps({"name": name}),
+    )
+
+    yield mocked_response
+
+
+class TestTwinCreateInstance(object):
+    @pytest.fixture
+    def service_client_with_retry(self, mocked_response, start_twin_response):
+        mocked_response.add(
+            method=responses.PUT,
+            url=re.compile(
+                "https://(.*)management.azure.com/subscriptions/(.*)/"
+                "resourceGroups/{}/providers/Microsoft.DigitalTwins/digitalTwinsInstances/{}".format(
+                    resource_group_name, name
+                )
+            ),
+            body=provisioning,
+            status=201,
+            headers={
```

Comment:
We could add the `Retry-After` header to the mock to ensure header value is being honored.